src/main/java/software/amazon/awssdk/crt/auth/credentials/ProfileCredentialsProvider.java

```diff
@@ -0,0 +1,133 @@
+         * @param clientBootstrap client bootstrap to use
+         * @return The current builder
+         */
+        public ProfileCredentialsProviderBuilder withClientBootstrap(ClientBootstrap clientBootstrap) {
+            this.clientBootstrap = clientBootstrap;
+
+            return this;
+        }
+
+        ClientBootstrap getClientBootstrap() { return clientBootstrap; }
+
+        /**
+         * Sets the tls context to use for any secure network connections made while sourcing credentials.
+         * @param tlsContext the tls context to use when establishing network connections
+         * @return The current builder
+         */
+        public ProfileCredentialsProviderBuilder withTlsContext(TlsContext tlsContext) {
+            this.tlsContext = tlsContext;
+            return this;
+        }
+
+        TlsContext getTlsContext() { return tlsContext; }
+
+        /**
+         * Sets the name of the profile to use. If none is specified, the profile named "default" is used.
+         * @param profileName the profile name to use
+         * @return The current builder
+         */
+        public ProfileCredentialsProviderBuilder withProfileName(String profileName) {
+            this.profileName = profileName;
+            return this;
+        }
+
+        String getProfileName() { return profileName; }
+
+        /**
+         * Sets the name of the config file to use. If none is specified, a name of "~/.aws/config" (Linux and Mac) or
+         * "%USERPROFILE%\.aws\config" (Windows) is used.
+         * @param configFileNameOverride the config file name to use
+         * @return The current builder
+         */
+        public ProfileCredentialsProviderBuilder withConfigFileNameOverride(String configFileNameOverride) {
+            this.configFileNameOverride = configFileNameOverride;
+            return this;
+        }
+
+        String getConfigFileNameOverride() { return configFileNameOverride; }
+
+        /**
+         * Sets the name of the credentials file to use. If none is specified, a name of "~/.aws/credentials" (Linux and
+         * Mac) or "%USERPROFILE%\.aws\credentials" (Windows) is used.
+         * @param credentialsFileNameOverride the credentials file name to use
+         * @return The current builder
+         */
+        public ProfileCredentialsProviderBuilder withCredentialsFileNameOverride(String credentialsFileNameOverride) {
+            this.credentialsFileNameOverride = credentialsFileNameOverride;
+            return this;
+        }
+
+        String getCredentialsFileNameOverride() { return credentialsFileNameOverride; }
+
+        public ProfileCredentialsProvider build() { return new ProfileCredentialsProvider(this); }
+    }
+
+    private ProfileCredentialsProvider(ProfileCredentialsProviderBuilder builder) {
+        super();
+
+        long nativeHandle = profileCredentialsProviderNew(
+                this,
+                toNativeHandle(builder.clientBootstrap),
```

Comment:
client bootstrap and tls context need to be required I think.  You can't do role assumption without them.

Reply:
It appears the underlying code [utilizes a default TLS context](https://github.com/awslabs/aws-c-auth/blob/a5dbb0cb7fccb6405b16245f4be7cbd2225496bd/source/credentials_provider_profile.c#L250-L258) if none is provided (but doesn't do the same for client bootstrap).

It also appears that role assumption only occurs [when the `role_arn` key is found in the profile](https://github.com/awslabs/aws-c-auth/blob/a5dbb0cb7fccb6405b16245f4be7cbd2225496bd/source/credentials_provider_profile.c#L400-L407).

Do we want to require a bootstrap even if the profile isn't guaranteed to contain `role_arn`? If so, is there a sensible default for the Java builder to use?

Reply:
Technically creating a new ClientBootstrap(null, null) will use a shared, lazily created host resolver and event loop group, so that may be a possibily.  Failing creation if it's not provided is also a possibility.

Reply:
Using `new ClientBootstrap(null, null)` makes sense. Will update!